web/shutdown.go

```diff
@@ -0,0 +1,67 @@
+package web
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+)
+
+// Server will create a http.Server from the Go standard library
+type Server struct {
+	Addr    string        // TCP address to listen on, ":http" if empty
+	Handler http.Handler  // handler to invoke, http.DefaultServeMux if nil
+	Timeout time.Duration // Timeout to wait for shutdown
+}
+
+// ListenAndServe mirrors the function from the Go standard library
+// ListenAndServe always returns a non-nil error
+func (s *Server) ListenAndServe() error {
+	server := http.Server{
+		Addr:    s.Addr,
+		Handler: s.Handler,
+	}
+
+	// We want to report the listener is closed.
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	done := make(chan struct{})
+	var output error
```

Comment:
why not call it `err`?

Reply:
Later lines set `output = err` to output those errors instead.

https://github.com/moexmen/gokit/blob/add-graceful-shutdown/web/server.go#L56
https://github.com/moexmen/gokit/blob/add-graceful-shutdown/web/server.go#L59

Reply:
Assuming server starts up correctly, would it be a race condition to see who sets `output` or will `ListenAndServe` always be the last to set `output`?

Reply:
That's true, it'll be a race condition. I didn't handle the data concurrency problem.